shared/engine/transport-shared.js

```diff
@@ -123,7 +123,10 @@ class TransportShared extends RobustTransport {
       // delay the call back to us
       const handler = payload => {
         this._injectInstrumentedResponse(payload)
-        incomingRPCCallback(payload)
+        // Defer a frame since decoding can take awhile, using setTimeout and not setImmediate on purpose
+        setTimeout(() => {
```

Comment:
i didn't want it to change behavior that much, we could evaluate it but i was worried at the time about it being dramatically out of order , etc. So far we've been doing a lot of request idle for calls we don't really care about and things like logging, this would affect all rpcs. we can try it for sure